Libs/CMakeLists.txt

```diff
@@ -59,6 +59,27 @@ if(Slicer_BUILD_CLI_SUPPORT)
   set(ModuleDescriptionParser_USE_PYTHON OFF)
 endif()
 
+set(MRML_APPLICATION_NAME "${Slicer_MAIN_PROJECT_APPLICATION_NAME}")
+
+# Convert version to 0xXXYYZZ format where XX=major, YY=minor, ZZ=patch version in hexadecimal.
+# Add a leading 1 (256*256*256*1) to force having a leading 0 for major version (e.g., 04 instead of just 4)
+math(EXPR MRML_VERSION_HEX "256*256*256*1 + 256*256*${Slicer_MAIN_PROJECT_VERSION_MAJOR}
+  + 256*${Slicer_MAIN_PROJECT_VERSION_MINOR} + ${Slicer_MAIN_PROJECT_VERSION_PATCH}" OUTPUT_FORMAT HEXADECIMAL)
```

Comment:
Could also format like this:

```
math(EXPR MRML_VERSION_HEX "1 << 24 
  | ${Slicer_MAIN_PROJECT_VERSION_MAJOR} << 16
  | ${Slicer_MAIN_PROJECT_VERSION_MINOR} << 8 
  | ${Slicer_MAIN_PROJECT_VERSION_PATCH}"
  OUTPUT_FORMAT HEXADECIMAL)
```

Inspired from what is done for python in [cpython/Include/patchlevel.h](https://github.com/python/cpython/blob/c49016e67c3255b37599b354a8d7995d40663991/Include/patchlevel.h#L29-L35)

Reply:
To be consistent with the new style of the form `Slicer 4.4.0 123456`, should also include `123456` in `MRML_VERSION_HEX` ?

Reply:
This version is used in C precompiler macros (for example `#if MRML_SUPPORT_VERSION < 0x040000`). If we added the revision then this would not work.

We could use the revision _instead_ of major.minor.patch, but probably we don't need this granularity and it would be bigger change without much (or any) added value.

Reply:
> used in C precompiler macros (for example #if MRML_SUPPORT_VERSION < 0x040000)

I would prefer the use a macro like `MRML_VERSION_CHECK` because as soon as the minor version go beyond 9, it would be very easy to introduce wrong comparison (either in the main code base or in extension). For example, `MRML_SUPPORT_VERSION < 0x040b00` should be used instead of `MRML_SUPPORT_VERSION < 0x041100`

Reply:
> We could use the revision instead of major.minor.patch, but probably we don't need this granularity and it would be bigger change without much (or any) added value.

Since MRML version  is now coupled with  the Slicer version, after introducing a change into MRML scene, we would be able to test for the version only after the Slicer version is updated.

Instead of appending the Slicer revision, what about using a scheme like `MRML X.Y.Z.M` where:
* X -> Slicer major
* Y -> Slicer minor
* Z -> Slicer patch
* M -> MRML patch

This would allow us to bump the MRML version  while doing development without having to  bump the Slicer version.

Reply:
It's all good for minor and patch version up to 255 (0x... will be compared as a hexadecimal number), but I agree that MRML_VERSION_CHECK() macro would be probably more familiar for developers and it is nicer if we don't expose hex representation in public CMake variables and #defined values.

Reply:
How would you set MRML patch? From the revision? We currently allow either the git hash or the commit-count-based counter to be used as revision. Git has would not allow comparison. Should we remove the option of using git has as revision and only allow revision to be an integer?

Reply:
MRML patch would be an integer set manually in CMake (e.g in a `MRMLVersion.cmake`)

or it could be set by extracting the commit count using a command  like `git rev-list HEAD --count -- ./Libs/MRML/`

Reply:
MRML version is an application level information (many MRML nodes are outside Libs/MRML). Should we use main application revision?

Reply:
To answer the question, we need to define what is the meaning of "MRML Version". 

(1) Is it the version of the MRML Library ?

or

(2) Is it the version of the MRML Library and the description of the required MRML extensions needed to successfully read the scene ? 

I was assuming it was (1) and i was assuming we were conveniently setting the MRML version based on the Slicer version only because the corresponding sources were inside the Slice source tree. 

If we go with the definition (2), the version saved in the file would have to list the modules (and the their version) required to successfully load the scene.